devs/docs/release.rst

```diff
@@ -105,5 +105,21 @@ archived.
 You can archive releases by selecting *Edit*, unselecting the *Active*
 checkbox, and then saving.
 
+Release Notes 
+=======================
+
+Once a release is done, we create a release note, which is
+
+- Added to the documentation
+- Sent out via mail as testing release note
```

Comment:
This is already documented in our internal checklist for releases. I wouldn't duplicate that here.